fuel-p2p/src/orchestrator.rs

```diff
@@ -0,0 +1,98 @@
+use fuel_core_interfaces::model::Vote;
+use fuel_tx::Transaction;
+use libp2p::identity::Keypair;
+use tokio::sync::mpsc::{Receiver, Sender};
+
+use crate::{
+    behavior::FuelBehaviourEvent,
+    config::P2PConfig,
+    gossipsub::messages::GossipsubMessage,
+    service::{FuelP2PEvent, FuelP2PService},
+};
+
+pub struct NetworkOrchestrator {
+    p2p_service: FuelP2PService,
+
+    // receivers
+    rx_block_producer: Receiver<()>,
+    rx_bft: Receiver<Vote>,
+
+    // senders
+    tx_synchronizer: Sender<()>,
+    tx_transaction_pool: Sender<Transaction>,
+    tx_bft: Sender<()>,
+}
+
+impl NetworkOrchestrator {
+    pub async fn new(
+        local_keypair: Keypair,
+        p2p_config: P2PConfig,
+        rx_block_producer: Receiver<()>,
+        rx_bft: Receiver<Vote>,
+
+        tx_synchronizer: Sender<()>,
+        tx_transaction_pool: Sender<Transaction>,
+        tx_bft: Sender<()>,
+    ) -> Self {
+        let p2p_service = FuelP2PService::new(local_keypair, p2p_config)
+            .await
+            .unwrap();
+
+        Self {
+            p2p_service,
+
+            rx_bft,
+            rx_block_producer,
+
+            tx_bft,
+            tx_synchronizer,
+            tx_transaction_pool,
+        }
+    }
+
+    pub async fn run(&mut self) {
+        loop {
+            tokio::select! {
+                p2p_event = self.p2p_service.next_event() => {
+                    if let FuelP2PEvent::Behaviour(behaviour_event) = p2p_event {
+                        match behaviour_event {
+                            FuelBehaviourEvent::GossipsubMessage { message, topic_hash, peer_id } => {
+                                match message {
+                                    GossipsubMessage::NewTx(tx) => {
+                                        let _ = self.tx_transaction_pool.send(tx);
+                                    },
+                                    GossipsubMessage::NewBlock(block) => {
+
+                                    },
+                                    GossipsubMessage::ConensusVote(vote) => {
+
+                                    },
+                                }
+                            },
+                            FuelBehaviourEvent::RequestMessage { request_message, request_id } => {
```

Comment:
Here we receive a request message from another peer,
currently this is only a request for a FuelBlock,

I need to get the Block and respond back.

From where do I fetch it?

Reply:
`fuel-txpool` and `fuel-relayer` take the trait that db needs to implement, and with that, you can directly access the database to fetch block body/header.

Reply:
I think one thing is missing, the channel that `fuel-sync` is going to use to request blocks from the network.

Maybe it will be cleaner for discussion to model channels in this format: https://github.com/FuelLabs/fuel-core/wiki/Interfaces-between-fuel-client-modules#p2p-channels
and then just apply changes to the wiki afterwards.